test/e2e/reconcile_objects_test.go

```diff
@@ -0,0 +1,185 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/openshift/cluster-monitoring-operator/pkg/manifests"
+	"github.com/openshift/cluster-monitoring-operator/test/e2e/framework"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/yaml"
+)
+
+func extractNamespacedNameFromFile[T metav1.ObjectMetaAccessor](t *testing.T, file string) types.NamespacedName {
+	t.Helper()
+
+	data, err := os.ReadFile("../../assets/" + file)
+	require.NoError(t, err)
+
+	var s T
+	require.NoError(t, yaml.Unmarshal(data, &s))
+	return types.NamespacedName{
+		Name:      s.GetObjectMeta().GetName(),
+		Namespace: s.GetObjectMeta().GetNamespace(),
+	}
+}
+
+func TestSecretsReconciliation(t *testing.T) {
+	// List of secrets that should not be synced during operator's reconciliation.
+	var (
+		namespaceMonitoring                 = f.Ns
+		namespaceUserWorkloadMonitoring     = f.UserWorkloadMonitoringNs
+		extractNamespacedNameFromFileSecret = extractNamespacedNameFromFile[*v1.Secret]
+	)
+	unsyncedSecrets := []types.NamespacedName{
+		extractNamespacedNameFromFileSecret(t, manifests.AlertmanagerConfig),
```

Comment:
I think it's simpler/more readable/less error prone is we just hardcode the names here

Reply:
we don't add new secrets regularly, it's always better to minimize risks that the test itself is broken.
Having hardcoded names is more readable (no need to go refer to the manifest to get the name of a secret) and we don't have to worry that the "fetch secret code" is broken (I'm not saying it's broken now :))

We can always change in the future if we start adding/removing secrets regularly in the future.